pkg/cmd/operator/analyze.go

```diff
@@ -0,0 +1,98 @@
+package operator
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/errors"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+)
+
+type AnalyzeOptions struct {
+	genericclioptions.ClientConfig
+	ArchivePath string
+}
+
+func NewAnalyzeOptions(streams genericclioptions.IOStreams) *AnalyzeOptions {
+	return &AnalyzeOptions{
+		ClientConfig: genericclioptions.NewClientConfig("scylla-analyze"),
+	}
+}
+
+func NewAnalyzeCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewAnalyzeOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "analyze",
+		Short: "Run scylla-operator analyze.",
+		Long:  "Run the scylla anallyze to analyze either live cluster or gathered data",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run(streams, cmd)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *AnalyzeOptions) AddFlags(cmd *cobra.Command) {
+	o.ClientConfig.AddFlags(cmd)
+
+	cmd.Flags().StringVarP(&o.ArchivePath, "archive-path", "", o.ArchivePath, "Path to compresed must-gather archive or directory having must-gather structure")
+}
+
+func (o *AnalyzeOptions) Validate() error {
+	var errs []error
+
+	errs = append(errs, o.ClientConfig.Validate())
+
+	_, err := os.Stat(o.ArchivePath)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("archive-path is not a valid filepath"))
+	}
+
+	if len(o.Kubeconfig) != 0 && len(o.ArchivePath) != 0 {
+		errs = append(errs, fmt.Errorf("kubeconfig and archive-path can't both be set"))
+	}
+
+	if len(o.Kubeconfig) == 0 && len(o.ArchivePath) == 0 {
+		errs = append(errs, fmt.Errorf("kubeconfig and archive-path can't both be empty"))
```

Comment:
empty kubeconfig is fine, a default is used in that case. If both are empty we implicitly assume working on a live cluster.

Reply:
removed this check